coffee/src/CoffeeMachineLoop.java

```diff
@@ -0,0 +1,184 @@
+        System.out.println(AMOUNT_OF_COFFEE + " coffee beans");
+        System.out.println(AMOUNT_OF_CUPS + " of disposable cups");
+        System.out.println("$" + AMOUNT_OF_MONEY + " of money");
+
+    }
+
+    /**
+     * Check for resources to make at list one cup of coffee.
+     *
+     * @param coffeeType Coffee type code (1 - espresso, 2 - latte, 3 - cappuccino).
+     *
+     * @return Has resources or not.
+     */
+    private static boolean hasResources(int coffeeType)
+    {
+        boolean has = true;
+
+        if(AMOUNT_OF_WATER < WATER_FOR_ONE_CUP[coffeeType]) {
+            System.out.println("Sorry, not enough water!");
+            has = false;
+        }
+
+        if(AMOUNT_OF_COFFEE < COFFEE_FOR_ONE_CUP[coffeeType]) {
+            System.out.println("Sorry, not enough coffee!");
+            has = false;
+        }
+
+        if(AMOUNT_OF_MILK < MILK_FOR_ONE_CUP[coffeeType]) {
+            System.out.println("Sorry, not enough milk!");
+            has = false;
+        }
+
+        if(AMOUNT_OF_CUPS == 0) {
+            System.out.println("Sorry, not enough cups!");
+            has = false;
+        }
+
+        return has;
+    }
+
+    private static void buy()
+    {
+        int coffeeType = readIntegerValue("What do you want to buy? 1 - espresso, 2 - latte, 3 - cappuccino:");
+        --coffeeType;
+
+        if (hasResources(coffeeType)) {
+            System.out.println("I have enough resources, making you a coffee!");
+
+            AMOUNT_OF_WATER -= WATER_FOR_ONE_CUP[coffeeType];
+            AMOUNT_OF_MILK -= MILK_FOR_ONE_CUP[coffeeType];
+            AMOUNT_OF_COFFEE -= COFFEE_FOR_ONE_CUP[coffeeType];
+            AMOUNT_OF_MONEY += MONEY_FOR_ONE_CUP[coffeeType];
+
+            AMOUNT_OF_CUPS--;
+        }
+    }
+
+    private static void fill()
+    {
+        AMOUNT_OF_WATER += readIntegerValue("Write how many ml of water do you want to add:");
+        AMOUNT_OF_MILK += readIntegerValue("Write how many ml of milk do you want to add:");
+        AMOUNT_OF_COFFEE += readIntegerValue("Write how many grams of coffee beans do you want to add:");
+        AMOUNT_OF_MONEY += readIntegerValue("Write how many disposable cups of coffee do you want to add:");
+    }
+
+    private static void take()
+    {
+        System.out.println("I gave you $" + AMOUNT_OF_MONEY);
+        AMOUNT_OF_MONEY = 0;
+    }
+
+    /**
+     * Reads an integer from standard input.
+     *
+     * @param msg Prompt message.
+     *
+     * @return Integer value.
+     */
+    private static int readIntegerValue(String msg)
+    {
+        System.out.println(msg);
+        System.out.print("> ");
+        return scan.nextInt();
+    }
+
+    /**
+     * Reads text from standard input.
+     *
+     * @param msg Prompt message.
+     *
+     * @return String value.
+     */
+    private static String readStringValue(String msg)
+    {
+        System.out.println(msg);
+        System.out.print("> ");
```

Comment:
@tatevikb why you need this ?

Reply:
if you mean while loop in this method, we need it because 'new line symbol' is also a symbol and `nextLine()` returns an empty line.

I created example for this test.

```
public class Test {
public static void main(String[] args)
{
Scanner scan = new Scanner(System.in);

int nm = scan.nextInt();

String st0 = scan.nextLine();
String st1 = scan.nextLine();
}
}
```